flake.nix

```
@@ -1,70 +1,92 @@
 {
   inputs.flake-utils.url = "github:numtide/flake-utils";
-  #inputs.nixpkgs.url = "github:NixOS/nixpkgs";
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs";
 
   outputs = { self, nixpkgs, flake-utils, ... }:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = nixpkgs.legacyPackages.${system};
         riscv-toolchain =
-#          (import <nixpkgs> {}).pkgsCross.riscv32;
-#          (import <nixpkgs> {}).pkgsCross.riscv64;
           import nixpkgs {
             localSystem = "${system}";
             crossSystem = {
               config = "riscv32-none-elf";
-              libc = "newlib";
-              abi = "ilp32";
+              libc = "newlib-nano";
+              abi = "ilp32d";
             };
           };
       in
         {
-          packages.fe310 = riscv-toolchain.stdenv.mkDerivation {
-            name = "riscv-fe310-firmware";
+          packages = {
 
-            src = ./.;
+            fe310 = riscv-toolchain.stdenv.mkDerivation {
+              name = "riscv-fe310-firmware";
 
-            buildInputs = with pkgs; [
-              riscv-toolchain.buildPackages.gcc
-              riscv-toolchain.buildPackages.binutils.bintools
-            ];
+              src = ./.;
 
-            buildPhase = ''
-              make -C fw/fe310
-            '';
+              buildInputs = with pkgs; [
+                riscv-toolchain.buildPackages.gcc
+              ];
 
-            installPhase = ''
-              cp -r fw/fe310/libeos.a $out
-            '';
+              buildPhase = ''
+                make -C fw/fe310
+              '';
 
-            # TODO add check phase once we can build tests
-            # checkPhase = ''
-            #   make -C fw/fe310/test
-            # '';
+              installPhase = ''
+                cp -r fw/fe310/libeos.a $out
+              '';
 
-          };
+              postInstall = ''
+                mkdir -p $out
+                for file in ${riscv-toolchain.newlib-nano}/riscv32-none-elf/lib/*.a; do
+                ln -s $file $out
+                done
+                for file in $out/*.a; do
+                mv "$file" "''${file%%.a}_nano.a"                
```

Nitpick: indentation
```suggestion
                  mv "$file" "''${file%%.a}_nano.a"                
```